packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -318,7 +319,7 @@ export function getValidationErrors(config: any): string[] {
 
           const { privateKey, balance } = account;
 
-          validatePrivateKey(privateKey, HARDHAT_NETWORK_NAME, errors);
+          validatePrivateKey(privateKey, +index, HARDHAT_NETWORK_NAME, errors);
```

Comment:
If you use `accounts.entries` instead of `Object.entries(accounts)` in the for, `index` will be a number and it won't be necessary to make this cast.

Reply:
Oh nice